installer-clientside/install-gui/src/App.tsx

```diff
@@ -88,9 +95,23 @@ function InstallerGui(props: InstallerProps) {
               <Grid.Col span={6}>
                 <p><b>Which printer?</b></p>
                 <Select placeholder="Choose a printer"
-                        value={printerIp} onChange={updatePrinter}
-                        data={props.printersAvailable.map((p) => ({ value: p.ip, label: `${p.ip} (${p.serial})` }) )}
+                        value={printerIp} onChange={(v) => {
+                          if (v == "$manual-input") {
+                            setIsManualIp(true);
+                            setPrinterIp(null);
+                          } else {
+                            setIsManualIp(false);
+                            updatePrinter(v);
+                          }
+                        }}
+                        data={[
+                          ...props.printersAvailable.map((p) => ({ value: p.ip, label: `${p.ip} (${p.serial})` }) ),
+                          { value: "$manual-input", label: "Manually enter address..." }
+                        ]}
                         disabled={props.isConnecting} />
+                { isManualIp && <span>
+                  <TextInput placeholder="192.168.1.XXX" value={printerIp} onChange={(ev) => updatePrinter(ev.currentTarget.value)} disabled={props.isConnecting}/>
```

Comment:
on US DOCSIS networks, this is often 10.1.x.x!  maybe:
```suggestion
                  <TextInput placeholder="aa.bb.cc.dd" value={printerIp} onChange={(ev) => updatePrinter(ev.currentTarget.value)} disabled={props.isConnecting}/>
```